shell/platform/windows/win32_window.cc

```diff
@@ -0,0 +1,207 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/windows/win32_window.h"
+
+namespace flutter {
+
+Win32Window::Win32Window() {
+  dpi_helper_ = std::make_unique<Win32DpiHelper>();
+
+  // Assume Windows 10 1607 for DPI handling.
+  // TODO handle DPI robustly downlevel
+  // TODO the calling applicaiton should participate in setting the DPI
+  // awareness mode
+  auto result = dpi_helper_->SetProcessDpiAwarenessContext(
+      DPI_AWARENESS_CONTEXT_PER_MONITOR_AWARE_V2);
+
+  if (result != TRUE) {
+    OutputDebugString(L"Failed to set PMV2");
+  }
+}
+Win32Window::~Win32Window() {
+  Destroy();
+}
+
+void Win32Window::Initialize(const char* title,
+                             const unsigned int x,
+                             const unsigned int y,
+                             const unsigned int width,
+                             const unsigned int height) {
+  Destroy();
+  auto converted_title = NarrowToWide(title);
+
+  auto window_class = ResgisterWindowClass(converted_title);
+
+  CreateWindow(window_class.lpszClassName, converted_title.c_str(),
+               WS_OVERLAPPEDWINDOW | WS_VISIBLE, x, y, width, height, nullptr,
+               nullptr, window_class.hInstance, this);
+}
+
+std::wstring Win32Window::NarrowToWide(const char* source) {
+  auto length = strlen(source);
+  size_t outlen = 0;
+  std::wstring wideTitle(length, L'#');
+  mbstowcs_s(&outlen, &wideTitle[0], length + 1, source, length);
+  return wideTitle;
+}
+
+WNDCLASS Win32Window::ResgisterWindowClass(std::wstring& title) {
+  window_class_name_ = title;
+
+  WNDCLASS window_class{};
+  window_class.hCursor = LoadCursor(nullptr, IDC_ARROW);
+  window_class.lpszClassName = title.c_str();
+  window_class.style = CS_HREDRAW | CS_VREDRAW;
+  window_class.cbClsExtra = 0;
+  window_class.cbWndExtra = 0;
+  window_class.hInstance = GetModuleHandle(nullptr);
+  window_class.hIcon = nullptr;
+  window_class.hbrBackground = 0;
+  window_class.lpszMenuName = nullptr;
+  window_class.lpfnWndProc = WndProc;
+  RegisterClass(&window_class);
+  return window_class;
+}
+
+LRESULT CALLBACK Win32Window::WndProc(HWND const window,
+                                      UINT const message,
+                                      WPARAM const wparam,
+                                      LPARAM const lparam) noexcept {
+  if (message == WM_NCCREATE) {
+    auto cs = reinterpret_cast<CREATESTRUCT*>(lparam);
+    SetWindowLongPtr(window, GWLP_USERDATA,
+                     reinterpret_cast<LONG_PTR>(cs->lpCreateParams));
+
+    auto that = static_cast<Win32Window*>(cs->lpCreateParams);
+
+    // Since we are running in Per-monitor V2 mode, turn on automatic titlebar
```

Comment:
Avoid "we" in comments; I don't know who "we" is here. Particularly since this is library code.